Add Node.js guide (#226)
memgraph/documentation: pages/client-libraries/nodejs.mdx

correctly. The 7444 port enables Memgraph Lab to access and preview the logs, 
which is why both of these ports need to be exposed. 

For more information visit the getting started guide on 
[how to run Memgraph with Docker](/getting-started/install-memgraph/docker).

### Install Node.js 

Install the newest version of Node.js. You can find instructions on how to 
download and install Node.js suitable for your environment on the 
[official Node.js website](https://nodejs.org/en/download).

### Install JavaScript Driver

To create the following Node.js application, you first need to install the 
JavaScript Driver. 

Run the following *npm* command to install the [Neo4j JavaScript 
Driver](https://www.npmjs.com/package/neo4j-driver):

```
npm i neo4j-driver
```

<Callout type="info">

The Driver requires `npm` and any LTS version of Node.js.

</Callout>

### Set up your script

We'll be using *Express.js* to demonstrate how to connect to a running Memgraph
instance. [Express.js](https://expressjs.com/) is a web application framework 
that enables us to create complete Node.js applications. 

Let's jump in and connect a simple program to Memgraph.

> - Create a new directory for your application and position yourself in it:

Author: ```suggestion
Create a new directory for your application and position yourself in it:
```

```
mkdir MyApp
cd MyApp
```

> - Initialize your project and create a `package.json` file with the command:

Author: ```suggestion
Initialize your project and create a `package.json` file with the command:
```

```
npm init
```

- Install *Express.js* and the *Bolt driver* in the `/MyApp` directory while 
adding them to the dependencies list:

```
npm install express --save
npm install neo4j-driver --save
```

> - To make the actual program, create an `example.js` file and add the following

Author: ```suggestion
To make the actual program, create an `example.js` file and add the following
```

Memgraph uses Bolt protocol to communicate with the client using the exposed 
7687 port. Memgraph Lab is a web application you can use to visualize the data. 
It's accessible at http://localhost:3000 if Memgraph Platform is running 
correctly. The 7444 port enables Memgraph Lab to access and preview the logs, 
which is why both of these ports need to be exposed. 

For more information visit the getting started guide on 
[how to run Memgraph with Docker](/getting-started/install-memgraph/docker).

### Install Node.js 

Install the newest version of Node.js. You can find instructions on how to 
download and install Node.js suitable for your environment on the 
[official Node.js website](https://nodejs.org/en/download).

### Install JavaScript Driver

To create the following Node.js application, you first need to install the 
JavaScript Driver. 

Run the following *npm* command to install the [Neo4j JavaScript 
Driver](https://www.npmjs.com/package/neo4j-driver):

```
npm i neo4j-driver
```

<Callout type="info">

The Driver requires `npm` and any LTS version of Node.js.

</Callout>

### Set up your script

> We'll be using *Express.js* to demonstrate how to connect to ad running Memgraph

Reviewer: ad? what?

Author: a

> We'll be using *Express.js* to demonstrate how to connect to ad running Memgraph

Author: ```suggestion
We'll be using *Express.js* to demonstrate how to connect to a running Memgraph
```

instance. [Express.js](https://expressjs.com/) is a web application framework 
that enables us to create complete Node.js applications. 

Let's jump in and connect a simple program to Memgraph.

> - Create a new directory for your application and position yourself in it:

Reviewer: wouldn't this be better with steps?

> - Create a new directory for your application and position yourself in it:

Reviewer: DOn't know what i think about this bullets... i don't like them :D

  }

  res.send("Visualize your data in Memgraph Lab.")
}
catch (error) {
  console.error(error);
} finally {
  session.close();
}
```

The provided code snippet matches the node with the property `title` being 
"Regional Manager", runs a for loop through all the retrieved results and 
processes them. Assuming that the matched node has properties `id`, `name` and 
`title`, the output should be similar to this:

```
Id: 0, Name: Michael, Title: Regional Manager
```

#### Process the Relationship result

You can also receive a relationship information from a query. Let's say we want 
to retrieve from the database all of the relationships with type `WORKS_FOR`:

```js
const session = driver.session();
try{
  // Read a relationship between two nodes with 'WORKS_FOR' type
  result = await session.executeRead(async tx => { 
    return await tx.run(`  
      MATCH (a:Node)-[r:WORKS_FOR]->(b:Node) RETURN r;
      `
    )
  })
  
  // Process the results
  for(let record of result.records) { 
    console.log(record.get('r'))
  }

  res.send("Visualize your data in Memgraph Lab.")
}
catch (error) {
  console.error(error);
} finally {
  session.close();
}
```

> With the `console.log(record.get('r'))` line of code, you are able to get the 
> all of the information about certain relationship as your output. If you're only 
> interested in, for example,  relationship type, ID of the start node and ID of 
> the end node, you can replace that line of code with this:

Reviewer: ```suggestion
The `console.log(record.get('r'))` functions retrieve
all of the information about certain relationships. If you're only 
interested in, for example, the relationship type, ID of the start node and ID of 
the end node, you can replace that line of code with the following code:
```

```
console.log("Relationship type: ", record.get('r').type)
console.log("Start node ID: ", record.get('r').startNodeElementId)
console.log("End node ID: ", record.get('r').endNodeElementId)
```

> and receive an output similar to this:

Reviewer: ```suggestion
It will produce the following output:
```

```
Relationship type:  USES
Start node ID:  <ID>
End node ID:  <ID>
```

#### Process the Path result

You can receive paths from the database. Let's say we want 
to retrieve all of the paths containing the connections of user named `Michael`.
Since we are only reading from the database, we're going to use the `executeRead` 
method:

```js
const session = driver.session();
try{
  result = await session.executeRead(async tx => { 
    return await tx.run(`  
      MATCH p=(:Developer {name:'Michael'})-[]->() RETURN p
      `
    )
  })
  
  // Process the results
  for(let record of result.records) { 
    console.log(record.get('p'))
  }

  res.send("Visualize your data in Memgraph Lab.")
}
catch (error) {
  console.error(error);
} finally {
  session.close();
}
```

Path will contain [Nodes](#process-the-node-result) and 
[Relationships](#process-the-relationship-result), that can be accessed in 
the same way as in the previous examples, by casting them to the relevant type.

> 

Reviewer: ```suggestion
```
